xdsl/traits.py

```diff
@@ -236,6 +253,16 @@ def verify(self, op: Operation) -> None:
             )
 
 
+class OptionalSymbolOpInterface(SymbolOpInterface):
+    """
+    Helper interface specialization for an optional Symbol.
+    """
+
+    @staticmethod
+    def is_optional_symbol(op: Operation) -> bool:
+        return True
+
+
```

Comment:
Digging in MLIR's sources, that's the only specialization I saw, and a few times, hence exposing it directly as is in the xDSL practical/firendly mindset!